core/vm/eips.go

```diff
@@ -114,7 +114,7 @@ func enable1344(jt *JumpTable) {
 
 // opChainID implements CHAINID opcode
 func opChainID(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
-	chainId, _ := uint256.FromBig(interpreter.evm.chainConfig.ChainID)
+	chainId, _ := uint256.FromBig(interpreter.evm.chainRules.ChainID)
```

Comment:
Why the change? [Upstream](https://github.com/ethereum/go-ethereum/blob/a9523b6428238a762e1a1e55e46ead47630c3a23/core/vm/eips.go#L110) uses `chainConfig`.

Reply:
because this PR makes chainConfig an interface and we cannot reference a field on that. Open to other suggestions